ssv/proposer.go

```diff
@@ -153,25 +175,44 @@ func (r *ProposerRunner) ProcessPostConsensus(signedMsg *SignedPartialSignatureM
 		specSig := phase0.BLSSignature{}
 		copy(specSig[:], sig)
 
-		blk := &bellatrix.SignedBeaconBlock{
-			Message:   r.GetState().DecidedValue.BlockData,
-			Signature: specSig,
-		}
-		if err := r.GetBeaconNode().SubmitBeaconBlock(blk); err != nil {
-			return errors.Wrap(err, "could not submit to Beacon chain reconstructed signed Beacon block")
+		if r.decidedBlindedBlock() {
+			blk := &bellatrix2.SignedBlindedBeaconBlock{
+				Message:   r.GetState().DecidedValue.BlindedBlockData,
+				Signature: specSig,
+			}
+			if err := r.GetBeaconNode().SubmitBlindedBeaconBlock(blk); err != nil {
+				return errors.Wrap(err, "could not submit to Beacon chain reconstructed signed blinded Beacon block")
+			}
+		} else {
+			blk := &bellatrix.SignedBeaconBlock{
+				Message:   r.GetState().DecidedValue.BlockData,
+				Signature: specSig,
+			}
+			if err := r.GetBeaconNode().SubmitBeaconBlock(blk); err != nil {
+				return errors.Wrap(err, "could not submit to Beacon chain reconstructed signed Beacon block")
+			}
 		}
 	}
 	r.GetState().Finished = true
 	return nil
 }
 
+// decidedBlindedBlock returns true if decided value has a blinded block, false if regular block
+// WARNING!! should be called after decided only
+func (r *ProposerRunner) decidedBlindedBlock() bool {
+	return r.BaseRunner.State.DecidedValue.BlindedBlockData != nil
```

Comment:
late msg could be received and will result in panic. i think we should add check for decided value here